main.py

```diff
@@ -187,43 +193,65 @@ def import_assets(self):
         # print(tmx_data.layers)
 
     def setup(self, tmx_maps, player_start_pos):
+        """ create tiles """
         islands = tmx_maps.get_layer_by_name("Islands")
         for x, y, surface in islands.tiles():
             # print(x * TILE_SIZE, y * TILE_SIZE, surface)
-            Tile(
+            src.sprites.Tile(
                 self.all_sprites,
                 pos=(x * TILE_SIZE, y * TILE_SIZE),
                 surf=surface,
             )
 
-    def run(self):
-
+    def run(self) -> None:
+        """ main loop of the game """
         while self.running:
-            for event in pygame.event.get():
-                if event.type == pygame.QUIT:
-                    exit()
+            self.handle_events()
+            self.update()
+            self.render()
+
+    def handle_events(self) -> None:
```

Comment:
How the game handles the player is a perfect starting point, good job on that. What I would like to see is instead of clicking on the screen which direction you go that you see options around the Player object and then move to that tile. Perhaps with a way of highlighting the tile you can go to. 

Things for later (no priority):
- So in future additions we can add a player turn functionality that checks if the player has moved that the turn can end.